generated/src/aws-cpp-sdk-dynamodb/source/DynamoDBClient.cpp

```diff
@@ -3117,60 +1546,66 @@ UpdateTimeToLiveOutcome DynamoDBClient::UpdateTimeToLive(const UpdateTimeToLiveR
-              auto endpointOutcome = DescribeEndpoints(endpointRequest);
-              if (endpointOutcome.IsSuccess() && !endpointOutcome.GetResult().GetEndpoints().empty())
-              {
-                  const auto& item = endpointOutcome.GetResult().GetEndpoints()[0];
-
-                  m_endpointsCache.Put(endpointKey, item.GetAddress(), std::chrono::minutes(item.GetCachePeriodInMinutes()));
-                  endpoint = Aws::String(SchemeMapper::ToString(m_clientConfiguration.scheme)) + "://" + item.GetAddress();
-                  AWS_LOGSTREAM_TRACE("UpdateTimeToLive", "Endpoints cache updated. Address: " << item.GetAddress() << ". Valid in: " << item.GetCachePeriodInMinutes() << " minutes. Making request to newly discovered endpoint.");
-                  endpointResolutionOutcome.GetResult().SetURI(endpoint);
-              }
-              else
-              {
-                  AWS_LOGSTREAM_ERROR("UpdateTimeToLive", "Failed to discover endpoints " << endpointOutcome.GetError() << "\n Endpoint discovery is not required for this operation, falling back to the regional endpoint.");
-                  endpointResolutionOutcome = endpointOutcome.GetError();
-              }
-          }
-      }
-      if (!enableEndpointDiscovery || !endpointResolutionOutcome.IsSuccess() || endpointResolutionOutcome.GetResult().GetURL().empty()) {
-          endpointResolutionOutcome = TracingUtils::MakeCallWithTiming<ResolveEndpointOutcome>(
-              [&]() -> ResolveEndpointOutcome { return m_endpointProvider->ResolveEndpoint(request.GetEndpointContextParams()); },
-              TracingUtils::SMITHY_CLIENT_ENDPOINT_RESOLUTION_METRIC,
-              *meter,
-              {{TracingUtils::SMITHY_METHOD_DIMENSION, request.GetServiceRequestName()}, {TracingUtils::SMITHY_SERVICE_DIMENSION, this->GetServiceClientName()}});
-      }
-      AWS_OPERATION_CHECK_SUCCESS(endpointResolutionOutcome, UpdateTimeToLive, CoreErrors, CoreErrors::ENDPOINT_RESOLUTION_FAILURE, endpointResolutionOutcome.GetError().GetMessage());
-      return UpdateTimeToLiveOutcome(MakeRequest(request, endpointResolutionOutcome.GetResult(), Aws::Http::HttpMethod::HTTP_POST, Aws::Auth::SIGV4_SIGNER));
+      return UpdateTimeToLiveOutcome(MakeRequestDeserialize(&request, request.GetServiceRequestName(), Aws::Http::HttpMethod::HTTP_POST, [&](Aws::Endpoint::AWSEndpoint& resolvedEndpoint) ->  void {
+        Aws::String endpointKey{"Shared"};
+        DescribeEndpointsRequest endpointRequest{};
+        this->UpdateEndpointWithDescribeEndpoints(resolvedEndpoint, request.GetServiceRequestName(), endpointKey, endpointRequest, false);
+      }));
     },
     TracingUtils::SMITHY_CLIENT_DURATION_METRIC,
     *meter,
     {{TracingUtils::SMITHY_METHOD_DIMENSION, request.GetServiceRequestName()}, {TracingUtils::SMITHY_SERVICE_DIMENSION, this->GetServiceClientName()}});
 }
 
+
+void DynamoDBClient::UpdateEndpointWithDescribeEndpoints(Aws::Endpoint::AWSEndpoint& resolvedEndpoint,
```

Comment:
nit: maybe something to indicate that its based on config? something like OptionallyUpdateDescribeEndpointsCache?

Reply:
good call